src/SoapCore/SoapEndpointMiddleware.cs

```diff
@@ -324,6 +325,10 @@ private async Task ProcessOperation(HttpContext httpContext, IServiceProvider se
 				{
 					status = StatusCodes.Status400BadRequest;
 				}
+				else if (ex?.Message == "The client has disconnected")
+				{
+					status = StatusCodes.Status400BadRequest;
+				}

```

Comment:
Error messages can change so it's best not to compare against those. 

```suggestion
				else if (ex is ConnectionResetException)
				{
					status = StatusCodes.Status400BadRequest;
				}

```

Using ConnectionResetException requires that we reference an additional nuget package but that's fine.

If you only want to check for the "The specified network name is no longer available"-exception you can do something like this:

`if(ex?.InnerException is COMException cex && cex.ErrorCode == unchecked((int)0x80070040))`

A more radical option would be to go even wider and just check for IOException. That is probably fine also

Reply:
I originally went with ConnectionResetException  but i am not sure which version to install since this supports multiple frameworks

Reply:
This is the issue 
![image](https://github.com/DigDes/SoapCore/assets/22575879/dc9cd364-989b-4d9c-ab5c-664c8e449a9c)

Reply:
This works for me (manually edit the .csproj file)
```
<ItemGroup Condition="$(TargetFramework) == 'netstandard2.0' OR $(TargetFramework) == 'netstandard2.1'">
	<PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
	<PackageReference Include="Microsoft.AspNetCore.Http" Version="2.1.34" />
	<PackageReference Include="Microsoft.AspNetCore.Mvc.Core" Version="2.1.38" />
	<PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="2.1.1" />
	<PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="2.1.1" />
	<PackageReference Include="System.IO.Pipelines" Version="4.7.0" />
	<PackageReference Include="System.CodeDom" Version="4.7.0" />
	<PackageReference Include="Microsoft.AspNetCore.Connections.Abstractions" Version="2.1.3" />
</ItemGroup>

<ItemGroup Condition="$(TargetFramework) == 'netcoreapp3.1'">
	<FrameworkReference Include="Microsoft.AspNetCore.App" />
	<PackageReference Include="System.CodeDom" Version="4.7.0" />
	<PackageReference Include="Microsoft.AspNetCore.Connections.Abstractions" Version="3.1.32" />
</ItemGroup>

<ItemGroup Condition="$(TargetFramework) == 'net5.0'">
	<FrameworkReference Include="Microsoft.AspNetCore.App" />
	<PackageReference Include="System.CodeDom" Version="5.0.0" />
	<PackageReference Include="Microsoft.AspNetCore.Connections.Abstractions" Version="5.0.17" />
</ItemGroup>
<ItemGroup Condition="$(TargetFramework) == 'net6.0'">
	<FrameworkReference Include="Microsoft.AspNetCore.App" />
	<PackageReference Include="System.CodeDom" Version="6.0.0" />
	<PackageReference Include="Microsoft.AspNetCore.Connections.Abstractions" Version="6.0.27" />
</ItemGroup>
```

Reply:
HResult can be used, but that one varies more

Reply:
I am making the csproj changes, give me 5  mins will push those

Reply:
Done, also incremented the nuget version so you could release